src/include/migraphx/matcher.hpp

```diff
@@ -646,6 +664,19 @@ MIGRAPHX_PRED_MATCHER(broadcast, instruction_ref ins)
     return contains({"broadcast", "multibroadcast"}, ins->name());
 }
 
+/*
+ * Input and output shapes have the same rank.  This is assumed for broadcast
+ * instructions for some fusions.
+ */
+MIGRAPHX_PRED_MATCHER(dims_match, instruction_ref ins)
+{
+    auto input_shape  = ins->inputs().front()->get_shape();
+    auto output_shape = ins->get_shape();
+    bool cond         = input_shape.dynamic() or output_shape.dynamic() or
```

Comment:
Dont need to check for dynamic shapes. We are only matching static shapes for the reduction fusions.

Reply:
I wrote a test and it matched, and caused an exception (before it got to this point). I did not keep the test because it's out of scope for this PR.  But right now there's nothing preventing these matchers from blowing up on dynamic inputs.

Reply:
> I did not keep the test because it's out of scope for this PR. But right now there's nothing preventing these matchers from blowing up on dynamic inputs.

Yes but @turneram is working on a more general solution to this in #2842, which will make these checks useless. So lets not complicate this matcher for now, and just focus on the current problem we are trying to solve.

Reply:
ok, changed.  (This code now resides in fuse_reduce.cpp)